documentation/CAMARA-Security-Interoperability.md

```diff
@@ -0,0 +1,158 @@
+# CAMARA Security and Interoperability Profile
+
+## Table of Contents
+<!-- TOC -->
+* Introduction
+* Audience
+* Conventions
+* General Considerations
+    * Transport Security
+* OIDC Authorization Code Flow
+    * Cross-Site Request Forgery Protection
+* Client-Initiated Backchannel Authentication Flow
+    * Authentication Request
+* Client Credentials Flow
+    * Access Token Request
+* The Scope Parameter
+* Client Authentication
+* References
+<!-- TOC -->
+
+
+## Introduction
+
+This document is the CAMARA Security and Interoperability Profile. To ensure interoperability and increased security, this technical specification restricts some options available in OIDC and CIBA, but does not change these standards.
+
+The CAMARA document sharpens the following for interoperability and security: 
+
+* Of the flows defined in OIDC CAMARA uses the [OIDC Authorization Code Flow](https://openid.net/specs/openid-connect-core-1_0.html#CodeFlowAuth)
+
+* Scope Parameter. Recommendations about the format of the scope parameter used in  CAMARA APIs.
+
+* Client Authentication. Specifications for client authentication within CAMARA.
+
+By encapsulating these elements within this document, it aims to provide a comprehensive guide for developers and operators, ensuring consistent implementation and adherence to standardized security measures across the CAMARA ecosystem. The defined OIDC profile not only facilitates the integration process, but also serves as a basic framework for developers wishing to leverage the CAMARA APIs while maintaining security and interoperability.
+
+
+## Audience
+
+The target audience for this document is the service/technical departments of operators exposing network functions via standard CAMARA APIs. And applications or client systems that consume CAMARA standard APIs to make use of the operator's network capabilities.
+
+
+## Conventions
+
+The keywords "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "NOT RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be interpreted as described in RFC 2119.
+
+## General Considerations
+
+### Transport Security
+All network connections MUST use TLS 1.2 or better.
+
+## OIDC Authorization Code Flow
+
+The OIDC Authorization Code Flow is defined in [OpenID Connect](https://openid.net/specs/openid-connect-core-1_0.html)
```

Comment:
I searched the numberverification repo for prompt and found this comment.

```text
    **Additional details:**

    - **(1):** Authentication must be automatic without any user interactions. Authentication methods such as SMS OTP or user/password are incompatible, as the goal is to validate the mobile phone number that is accessing the App. So it is required to be authentication via mobile network and without the user being involved. the use of parameter prompt=none, as described in **[OIDC Connect](https://openid.net/specs/openid-connect-core-1_0.html#AuthRequest)**, ensures no user interaction.

    - **(2):** The way in which the phone_number is retrieved depends on the implementation. For example, access token may be a self contained encrypted JWT, so API can decrypt and identify phone_number. Some other implementations might request the phone_number associated to the token from Authserver.
```

Why is the numberverification subproject using authorization code flow when "authentication must be automatic"?
Numberverification could use authorization code flow to collect user consent and then use the access token to call the numberverification API. If the access token is expired, and the client (mobile app or web app) has a refresh token, then the refresh token can be used to request a new access token. Whether the user is bothered again with login or consent collection is up to the AZ then. 
In authorization code flow the code request must be authenticated (user login). There can't be a prompt=none without client authentication. The AZ can and probably should reject a prompt=none request, if there is no indication that the request is "legitimate".

I have the impression that numberverification tries to misuse the OIDC prompt=none parameter. Which of course leads to all kind of trouble because the AZ might be abused by 3rd parties who use prompt=none to trick the AZ into leaking user PII (is the telephone number served by that telco?).

If the client is using the numberverification API with prompt=none then there MUST be some prove that the client is allowed to make that request.

My opinion: If the API subproject needs some specific AZ behavior then that behavior should be defined here and not in a subproject document.

We could add a section `prompt=none`here, that would fit the profile. But maybe numberverification is better served with sticking to authorization code flow and access tokens and refresh tokens, or if the API has a no-consent-needed variant then use OAuth2 client credentials flow.

@DT-DawidWroblewski What's your opinion?